packages/react-mutation-mapper/src/cache/DefaultStringQueryCache.ts

```diff
@@ -0,0 +1,33 @@
+import { action, observable } from 'mobx';
+
+import { CacheData, MobxCache } from '../model/MobxCache';
+
+export abstract class DefaultStringQueryCache<D>
+    implements MobxCache<D, string> {
+    protected _cache = observable.shallowMap<CacheData>();
+
+    @action
+    public get(query: string) {
+        if (!this._cache.get(query)) {
+            this._cache.set(query, { status: 'pending' });
+
+            this.fetch(query)
+                .then((d: D) =>
+                    this._cache.set(query, { status: 'complete', data: d })
+                )
+                .catch(() => this._cache.set(query, { status: 'error' }));
+        }
+
+        const data = this._cache.get(query);
+
+        return data ? data : null;
+    }
+
+    public fetch(query: string): Promise<D> {
+        return Promise.reject('No Default Implementation');
+    }
+
+    public get cache() {
+        return this._cache.toJS();
```

Comment:
The key is to prevent the cache to be updated?

Reply:
Previously we were returning the cache as is, but that is a `Map` instance, so it doesn't actually match the function signature, we need to return a proper JS object. Not sure if this is the best way to do it though.